modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py

```diff
@@ -1550,6 +1550,52 @@ def merge(df, lib, iterations, **kwargs):
                 iterations=i,
             )
 
+    def test_merge_float(self):
+        def merge(df, df2, on_columns, **kwargs):
+            return df.merge(df2, on=on_columns)
+
+        run_and_compare(
+            merge,
+            data={"A": [1, 2] * 1000},
+            data2={"A": [1.0, 3.0] * 1000},
+            on_columns="A",
+            force_lazy=False,
+        )
+
+    def test_merge_categorical(self):
+        def merge(df, df2, on_columns, **kwargs):
+            return df.merge(df2, on=on_columns)
+
+        run_and_compare(
+            merge,
+            data={"A": [1, 2] * 1000},
+            data2={"A": [1.0, 3.0] * 1000},
+            on_columns="A",
+            constructor_kwargs={"dtype": "category"},
+            comparator=lambda df1, df2: df1["A"].values == df2["A"].values,
```

Comment:
comparator is just being executed with an assumption that it raises an error if comparison fails:
https://github.com/modin-project/modin/blob/8817093cb2f16be3709513b1517d8a4bc679900c/modin/experimental/core/execution/native/implementations/hdk_on_native/test/utils.py#L321

we likely need an assertion here

Reply:
Sure, assert is missing here. Thanks!